src/plan.rs

```diff
@@ -124,11 +122,11 @@ async fn plan_query<T: Debug>(storage: &dyn Store<T>, query: Query) -> Result<Qu
                 selection,
                 group_by,
                 having,
+                order_by,
             };
 
             Ok(Query {
                 body: SetExpr::Select(Box::new(select)),
-                order_by: Vector::from(order_by).pop().0.into(),
```

Comment:
Here is the key of `index_order_by` failure.
If the planner succeeded to find the index in `ORDER BY`, then it needs to remove the last order_by item which is consumed by the index.

Reply:
Thanks.. I realized we should never remove previous logic while refactoring...